harvester/pmh_interface.py

```diff
@@ -16,12 +17,56 @@ def __init__(self, url):
         """
         self._sickle = Sickle(url)
 
-    def binding_ids(self, set_id):
+
+    def dc_identifiers(self, set_id):
         """
-        Iterate over all binding IDs in the given set.
+        Iterate over all DC identifiers in the given set.
 
         :param set_id: Set (also known as collection) identifier
         """
         records = self._sickle.ListRecords(metadataPrefix='oai_dc', set=set_id)
         for record in records:
-            yield record.header.identifier.split(':')[-1]
+            yield record.metadata['identifier'][0]
+
+    
+    def binding_id_from_dc(self, dc_identifier):
+        """
+        Parse binding ID from dc_identifier URL.
+
+        :param dc_identifier: DC identifier of a record
+        """
+        return dc_identifier.split("/")[-1]
+    
+
+    def fetch_mets(self, dc_identifier, folder_path, file_name=None):
+        """
+        Fetch METS as an XML document given a binding ID and save to disk.
+
+        :param dc_identifier: DC identifier of a record
+        :param folder_path: Path to folder to which the METS file will be stored
+        :param file_name: Name of the file to which the METS will be stored (optional parameter)
+        """
+
+        mets_url = f"{dc_identifier}/mets.xml?full=true"
+        xml_response = requests.get(mets_url)
+
+        if not file_name:
+            path = f'{folder_path}/{self.binding_id_from_dc(dc_identifier)}_METS.xml'
+        else:
+            path = f'{folder_path}/{file_name}'
+
+        open(path, "w").write(xml_response.text)
```

Comment:
It's best practice to also close files. In practice there usually aren't problems as Python3 garbage collector is pretty good at closing files it notices are no longer used in code, but in theory bad stuff can happen. For example in buffered writes (AFAIK the default way of writing to file) some content might not actually get written to disk if the buffer does not fill up and the file is not closed: see e.g. https://stackoverflow.com/questions/25070854/why-should-i-close-files-in-python.

Reply:
Of course, not sure why I didn't use a context manager. Fix coming!

Reply:
Looking good :+1: